### DISCUSSION TITLE
如何排除Waline中邮件系统的故障呢

### DISCUSSION THREAD

**Discussion by LxnChan** [walinejs/waline#1043] [Q&A] [answered]
Waline v240，tcb云开发部署，用环境变量设置了smtp服务器。

smtp服务器用了icloud和office365的都没有用，收不到邮件，在日志里面找了但是没有
![image](https://user-images.githubusercontent.com/46388610/169516379-90b3ab7e-60f9-45be-baf6-dc1a365f0191.png)

确认邮件的相关配置都正确。

设置的环境变量如下：
![image](https://user-images.githubusercontent.com/46388610/169516706-31eb4436-22d2-4887-9792-0e385cfd7f4c.png)

**Comment by elkan1788** [comment]
看上去貌似是少了个 `SMTP_SERVICE` 参数名称的配置，可以对照官方手册 [邮件通知](https://waline.js.org/guide/server/notification.html#%E9%82%AE%E4%BB%B6%E9%80%9A%E7%9F%A5) 再检查下.

支持的运营商列表有：[service.json](https://github.com/nodemailer/nodemailer/blob/master/lib/well-known/services.json)

**Reply by LxnChan** [reply]
但我这个相当于是自定义邮件服务提供商啊，自定义的也需要写吗

**Comment by AnzhiZhang** [comment, marked as answer]
`SMTP_SERVICE` 无需填写。看环境变量没有问题，建议使用 Python、Java 等方式发送邮件，排查是否为邮件服务商问题
